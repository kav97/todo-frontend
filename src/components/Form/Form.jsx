import { faCirclePlus } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { useState } from "react";
import './Form.scss'

const Form = ({ onSubmit }) => {
    const [text, setText] = useState("");

    const handleSubmit = (event) => {
        event.preventDefault();
        setText("");
        onSubmit(event.target[0].value);
    };

  return (
    <form className="text-field" onSubmit={handleSubmit}>
      <input
        className="text-field__input"
        type="text"
        aria-label="Write a new todo item"
        placeholder="Add your task here..."
        value={text}
        onChange={(event) => setText(event.target.value)}
      />
      <button
        className="text-field__button"
        type="submit"
        aria-label="Submit new task"
      >
        <FontAwesomeIcon icon={faCirclePlus} />
      </button>
    </form>
  );
}

export default Form;