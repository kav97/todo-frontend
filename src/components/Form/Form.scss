.text-field__counter {
    color: green;

    &--amber {
        color: orange;
    }

    &--red{
        color: red;
    }
}