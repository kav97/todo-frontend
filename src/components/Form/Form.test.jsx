import { render } from "@testing-library/react";
import Form from "./Form";

it("Should render a form to screen", () => {
    const { component } = render(<Form />);
  
    expect(component).toMatchSnapshot();
});