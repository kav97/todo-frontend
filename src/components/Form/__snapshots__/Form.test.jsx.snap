// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render a form to screen 1`] = `undefined`;
